### DISCUSSION TITLE
Display [PSCUSTOMOBJECT] Inner Hashtables more cleanly?

### DISCUSSION THREAD

**Discussion by MrCarter2959** [EvotecIT/PSWriteHTML#318] [Q&A] [answered]
I have a rather large collection of data that I'm storing in a List Object as a [PSCUSTOMOBJECT] with inner hashtables sorta like this

```Powershell
$myListObject.Add([PSCUSTOMOBJECT]@{
            Name = $($t.Name)
            Status = $($t.Status)
            Folder = @{
                    Status = $f.Config
                    Current = $t.Folder
                    Required = 'PATH SHOULD BE X'
                }
            CPUHotAdd =  @{
                    Status = $C.Status
                    Current = $t.cpu
                    Required = 'Enabled'
                }
```

How would you go about displaying them cleanly inside of a New-HTMLDatatable? I tried -FlattenObject which works, however it makes it a little confusing to read since it basically expands the Current Tab like Folder and adds Folder.Status. Folder.Current, Folder.Required. Is there a better way to go about show data like this?

**Comment by PrzemyslawKlys** [comment, marked as answer]
I've seen people doing nested tables somewhere in the issues/discussions but you would need to rewrite your object anyways. 

You could use TreeView functionality although I've not played a lot with it.

You could use linked tables to display subtables in different tables and link the data. 

You could just rewrite the object the way flattenobject does but manually so you have more control over names of columns

**Reply by MrCarter2959** [reply]
Rewriting the object isn't a big deal. If i wanted to use the nested tables, would there be a good example somewhere? Or idea that shows how the object should be built?

**Reply by PrzemyslawKlys** [reply]
As mentioned:
- https://github.com/EvotecIT/PSWriteHTML/issues/233

**Reply by MrCarter2959** [reply]
Thanks! I was looking at that one, just wasn't sure if there was something else I was missing out there!

Inside of nested tables, does the New-HTMLTableCondition work as normal? I'm noticing I'm not getting the New-HTMLTableConditions I've set. Everything else is working now! 👍 

```PowerShell
SubTable = [string]$(
                            New-HTMLTable -Title "Name" -DataTable $_.Object-HideFooter -AutoSize {
                                New-HTMLTableCondition -Name "Status" -ComparisonType string -Operator eq -Value 'VALUE' -BackgroundColor BrightRed -Color Black
                                New-HTMLTableCondition -Name "Status" -ComparisonType string -Operator eq -Value "OK" -BackgroundColor CaribbeanGreen -Color Black
                            } -InvokeHTMLTags
                        )
```

**Reply by PrzemyslawKlys** [reply]
I never used it myself to be honest, so never tested it. I am also not sure how/where the code for table conditions goes when those are nested.
